libs/parallelism/executors/include/hpx/executors/dataflow.hpp

```diff
@@ -176,17 +201,17 @@ namespace hpx { namespace lcos { namespace detail {
         ///////////////////////////////////////////////////////////////////////
         /// Passes the futures into the evaluation function and
         /// sets the result future.
-        HPX_FORCEINLINE
-        void execute(std::false_type, Futures&& futures)
+        template <typename Futures_>
+        HPX_FORCEINLINE void execute(std::false_type, Futures_&& futures)
```

Comment:
why the underscore (not that it matters, but it seems unusual for the typename) - aha, I think it is hiding another use of the same typename above

Reply:
Indeed, it's because the struct has a `Futures` template parameter. `Futures` is a tuple with the original arguments to `dataflow` whereas `Futures_` is a tuple which can have additional arguments inserted into it.